### PR TITLE
feat(HelmChart): Configurable securityContext in Pods

### PR DESCRIPTION
**Pull request by miloszbednarzak**
### SUMMARY
This PR introduces the possibility to customize securityContext in Kubernetes Pods, which is necessary if one wants to have compatibility with the Pod Security Standards.
see https://kubernetes.io/blog/2021/12/09/pod-security-admission-beta/

### TESTING INSTRUCTIONS
In order to test changes, one needs to change securityContext specification in `values.yaml` for any Pod and check if after deployment it changed desired outcome in deployed Pod.

### ADDITIONAL INFORMATION

- [x] Has associated issue: #20489
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by craig-rueda — APPROVED** [review]

**Comment by miloszbednarzak** [comment]
@craig-rueda should we do anything more until merging, do we need another review or can we proceed?

**Pull request merged**
